### PR TITLE
feat(file): client CRUDL

### PR DESCRIPTION
**Pull request by pilartomas**

### CODE REVIEW

File: src/api-types.ts [hunk 1]
```diff
@@ -329,6 +337,39 @@ export const HistoryOutputSchema = z.object({
       response: GenerateOutputSchema,
     }),
   ),
-  totalCount: z.number().int().min(0),
 });
 export type HistoryOutput = z.infer<typeof HistoryOutputSchema>;
+
+// FILES
+
+export const FilePurposeSchema = z.enum(['tune', 'template', 'other']);
```
Comment (pilartomas) [inline src/api-types.ts:345]:
```suggestion
export const FilePurposeSchema = z.enum(['tune', 'template']);
```

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -94,6 +95,7 @@
     "axios": "^1.3.4",
     "axios-cache-interceptor": "^1.0.1",
     "cross-fetch": "^3.1.6",
+    "form-data": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (pilartomas) [reply package.json:98]: CJS modules were dropped in `5.x`

File: src/client-types.ts [hunk 3]
```diff
@@ -1,6 +1,14 @@
 import { z } from 'zod';
 import * as ApiTypes from './api-types.js';
 import { Readable } from 'node:stream';
+import { FlagOption } from './helpers/types.js';
+
+// COMMON
+
+const ListSchema = z.object({
+  offset: z.number().int().nonnegative().nullish(),
+  count: z.number().int().positive().nullish(),
```
[the remark this answers is not in the document]
Comment (pilartomas) [reply src/client-types.ts:10]: This is not the resulting count but the desired count. It makes no sense to request zero items.

[the remark this answers is not in the document]
Comment (pilartomas) [reply src/client-types.ts:10]: I should rename this to `ListInputSchema`

File: src/helpers/types.ts [hunk 4]
```diff
@@ -1,2 +1,6 @@
 export type RequiredPartial<T, L extends keyof T> = Required<Pick<T, L>> &
   Partial<Omit<T, L>>;
+
+export type FlagOption<Key extends string, T extends boolean> = T extends true
+  ? { [k in Key]: true }
+  : { [k in Key]?: false };
```
[the remark this answers is not in the document]
Comment (pilartomas) [reply src/helpers/types.ts:6]: It doesn't allow for leaving out the key.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (pilartomas) [reply package.json:98]: That's the latest version afaik https://www.npmjs.com/package/form-data

File: src/api-types.ts [hunk 5]
```diff
@@ -1,6 +1,16 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
 import { z } from 'zod';
+import type FormData from 'form-data';
+
+// COMMON
+
+const PaginationOutputSchema = z.object({
+  totalCount: z.number().int().nonnegative(),
```
[the remark this answers is not in the document]
Comment (pilartomas) [reply src/api-types.ts:9]: I was hoping for nicer error messages during parsing but apparently it is just an alias.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tomas2D) [reply package.json:98]: I see; but maybe this package will be better https://www.npmjs.com/package/formdata-node

File: src/helpers/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Tomas2D) [reply src/helpers/types.ts:6]: I see.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (pilartomas) [reply package.json:98]: I considered that one but it dropped CJS support in `5.x`.

File: src/client-types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tomas2D) [reply src/client-types.ts:10]: Yeah, please rename it.

**Pull request merged**